core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -970,6 +963,48 @@ private void initAfterRebalance() {
             return new AbstractMap.SimpleImmutableEntry<>(bb, currentHandle);
         }
 
+        /**
+         * Advances next to higher entry without creating a ByteBuffer for key.
+         * Return previous index
+         */
+        Handle advanceStream(OakRKeyReferenceImpl key) {
+
+            if (state == null) {
+                throw new NoSuchElementException();
+            }
+
+            Chunk.State chunkState = state.getChunk().state();
+
+            if (chunkState == Chunk.State.RELEASED) {
+                initAfterRebalance();
+            }
+
+            if (key != null) state.getChunk().setKeyRefer(state.getIndex(),key);
+            Handle currentHandle = state.getChunk().getHandle(state.getIndex());
+            advanceState();
+            return currentHandle;
+        }
+
+        /**
+         * Advances next to higher key without creating a ByteBuffer for key.
+         * Do not get and return handle/value
+         */
+        void advanceStreamKeyOnly(OakRKeyReferenceImpl key) {
```

Comment:
I'm not sure why we need two almost identical methods here. At worst, we can use the same method with a single extra key reference instance in values stream iterator, but I think we can think of some abstraction that can save even this single instance and reduce duplication.

Reply:
I agree
for example line 1004 need to check if key!=null like 982
either remove the version that does not return handle or have a single method with a boolean to indicate whether or not to return handle.

Reply:
Left only one method